processing/src/main/java/org/apache/druid/query/policy/PolicyConfig.java

```diff
@@ -0,0 +1,144 @@
+
+  @JsonProperty
+  private final ImmutableList<String> allowedPolicies;
+
+  @JsonCreator
+  public PolicyConfig(
+      @JsonProperty("tablePolicySecurityLevel") TablePolicySecurityLevel tablePolicySecurityLevel,
+      @JsonProperty("allowedPolicies") ImmutableList<String> allowedPolicies
+  )
+  {
+    this.tablePolicySecurityLevel = tablePolicySecurityLevel;
+    this.allowedPolicies = allowedPolicies;
+  }
+
+  /**
+   * Returns a {@link PolicyConfig} instance that applies restrictions whenever seen fit, and allows all policy types.
+   */
+  public static PolicyConfig defaultInstance()
+  {
+    return new PolicyConfig(TablePolicySecurityLevel.APPLY_WHEN_APPLICABLE, ImmutableList.of());
+  }
+
+  /**
+   * Returns the table policy security level, higher value means stricter policy.
+   *
+   * @see TablePolicySecurityLevel
+   */
+  @JsonProperty
+  public TablePolicySecurityLevel getTablePolicySecurityLevel()
+  {
+    return tablePolicySecurityLevel;
+  }
+
+  /**
+   * Returns a list of allowed policy class.
+   * <p>
+   * An empty list means all policy classes are allowed.
+   */
+  @JsonProperty
+  public ImmutableList<String> getAllowedPolicies()
+  {
+    return allowedPolicies;
+  }
+
+  public boolean allowPolicy(@Nonnull Policy policy)
+  {
+    return allowedPolicies.isEmpty() || allowedPolicies.contains(policy.getClass().getSimpleName());
+  }
+
+  /**
+   * Returns true if the security level requires that, every table must have a policy during query execution stage,
+   * this means the table must have a non-empty value in the policy map.
+   */
+  public boolean policyMustBeCheckedAndExistOnAllTables()
+  {
+    return tablePolicySecurityLevel.securityLevel >= 2.0;
+  }
+
+  @Override
+  public String toString()
+  {
+    return "PolicyConfig{tablePolicySecurityLevel="
+           + tablePolicySecurityLevel
+           + ", allowedPolicies="
+           + allowedPolicies
+           + '}';
+  }
+
+  @Override
+  public boolean equals(Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    PolicyConfig that = (PolicyConfig) o;
+    return Objects.equals(tablePolicySecurityLevel, that.tablePolicySecurityLevel) &&
+           Objects.equals(allowedPolicies, that.allowedPolicies);
+  }
+
+  @Override
+  public int hashCode()
+  {
+    return Objects.hash(tablePolicySecurityLevel, allowedPolicies);
+  }
+
+
+  /**
+   * Defines how strict we want to enforce the policy on tables during query execution process.
+   * <ol>
+   *   <li>{@code APPLY_WHEN_APPLICABLE}, the most basic level, restriction is applied whenever seen fit.
+   *   <li>{@code POLICY_CHECKED_ON_ALL_TABLES_POLICY_MUST_EXIST}, every table must have a policy when requests come from external users.
+   * </ol>
+   */
+  public enum TablePolicySecurityLevel
+  {
+    @JsonProperty("0") APPLY_WHEN_APPLICABLE(0f),
```

Comment:
I believe this name is a bit unfortunate; I feel like this wants to be something like "allow all when not blocked"
if yes then it could be: `ALLOW_MISSING_POLICY`